### PR TITLE
[WIP] Influeza-like Illness (ILI) Criteria

### PR DESCRIPTION
**Pull request by smacpher**
Implements ETL to compute the Influenza-like Illness (ILI) Criteria (5A in the criteria [workbook](https://docs.google.com/spreadsheets/d/1q9yZaQ1_qwEecvYknS1sZsRbD6eDY7il_uIX2dInbr0/edit?pli=1#gid=0)).

CDC Criteria transformations will be implemented in a separate PR.

### CODE REVIEW

**Comment by lucasmbrown** [comment]
Hi @smacpher ! Awesome to have the data in. Here's the section from p. 35 of the [CDC document](https://drive.google.com/file/d/1zWLHPIz6UWiwrI5tYz1roy1t9PpqE0dj/view) about that criteria: 

Screen Shot 2020-06-04 at 9 04 12 AM

It looks like we'll need to update the [guidance spreadsheet](https://docs.google.com/spreadsheets/d/1q9yZaQ1_qwEecvYknS1sZsRbD6eDY7il_uIX2dInbr0/edit?pli=1#gid=0) -- @patricksheehan made an awesome first draft of this criteria but we can expand on it and turn it into all the detailed steps.

File: covid/extract.py [hunk 1]
```diff
@@ -44,3 +54,26 @@ def get_state_abbreviations_to_names():
...
+
+
+def extract_cdc_ili_data():
+    current_url = "https://gis.cdc.gov/grasp/flu2/PostPhase02DataDownload"
+    payload = {
+        "AppVersion": "Public",
+        "DatasourceDT": [{"ID": 0, "Name": "WHO_NREVSS"}, {"ID": 1, "Name": "ILINet"}],
+        "RegionTypeId": 5,
+        # Request all 59 regions.
+        "SubRegionsDT": [{"ID": i, "Name": i} for i in range(1, 60)],
```
Comment (lucasmbrown) [inline covid/extract.py:153]: Nice!

File: covid/extract_utils.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import zipfile
+from io import BytesIO
+
+
+def unzip_string(string):
```
Comment (lucasmbrown) [inline covid/extract_utils.py:5]: 👍 !

File: covid/transform.py [hunk 3]
```diff
@@ -518,6 +518,28 @@ def transform_covidtracking_data(df):
...
+    assert set(df["REGION TYPE"].unique()) == {"States"}
+
+    # Create a new column that combines the `YEAR` and `WEEK` column.
+    # Note: the `-1` sets these timestamp at Monday of each week.
+    df["_WEEK_TIMESTAMP"] = pd.to_datetime(
+        df["YEAR"] + "-W" + df["WEEK"] + "-1", format="%Y-W%W-%w"
+    )
+
+    # Create a new multi-index containing the state name (`REGION`) and timestamp of the week.
+    df = df.set_index(keys=["REGION", "_WEEK_TIMESTAMP"])
```
Comment (lucasmbrown) [inline covid/transform.py]: Nice! Eventually we might want to change the name of the `state` field to `STATE_SOURCE_FIELD` so we can merge with the rest of the large dataframe. But we can also sort that out in the next PR.

Comment (smacpher) [reply covid/transform.py]: Ahh yes, that makes sense – I'll add that here actually.

Comment (smacpher) [reply covid/transform.py]: Also question – should I use the year and week (1 - 52) to create start of week timestamps or end of week timestamps? Currently I'm translating eg. `Week 1 of 2020` to the first day (Monday) of the week. Maybe it makes sense to convert that to the last day (Sunday) of the week?

Comment (lucasmbrown) [reply covid/transform.py]: Ooh great question. Probably the last day of the week, yes -- that makes more sense with lining up the criteria.

**Comment by smacpher** [comment]
@lucasmbrown Ready for a final pass!

File: covid/extract.py [hunk 4]
```diff
@@ -47,6 +57,7 @@ def get_state_abbreviations_to_names():
     return abbreviations
 
 
+<<<<<<< HEAD
```
Comment (lucasmbrown) [inline covid/extract.py]: @smacpher -- still have some merge conflicts here!

File: covid/transform.py [hunk 5]
```diff
@@ -521,6 +521,33 @@ def transform_covidtracking_data(df):
...
+    # Drop the row containing the labels, and set a new multi-index using the region, year, and week.
+    df = df.iloc[1:].reset_index()
+
+    # Validate that the only region type is states to sanity check data.
+    assert set(df["REGION TYPE"].unique()) == {"States"}
+
+    # Create a new column that combines the `YEAR` and `WEEK` column.
+    # Note: The `-6` sets the date field to the start of the weekend (Saturday) for each week. We also subtract 1 week
+    #   to start the weeks at zero and ensure that they align with `https://www.epochconverter.com/weeks/2020`.
+    df[DATE_SOURCE_FIELD] = pd.to_datetime(
```
Comment (lucasmbrown) [inline covid/transform.py:540]: Is this converting to the last day (Saturday?) of the current week, as previously discussed? Or is it converting to the Saturday of the previous week?

**Review by lucasmbrown — CHANGES_REQUESTED** [review]
Looking p good, left a question and a cleanup comment!

File: covid/transform.py (hunk 5 above)
Comment (smacpher) [reply covid/transform.py:540]: I had to subtract 1 for some reason to get it to align with the linked week number tracker and Ryan's current sheet.

Comment (smacpher) [reply covid/transform.py:540]: I think because the CDC's week numbers are 1-indexed instead of 0-indexed

Comment (lucasmbrown) [reply covid/transform.py:540]: Makes sense!

**Review by lucasmbrown — APPROVED** [review]

File: covid/transform.py [hunk 6]
```diff
@@ -524,6 +524,33 @@ def transform_covidtracking_data(df):
...
+        df["YEAR"] + "-" + (df["WEEK"].astype(int) - 1).astype(str) + "-6",
+        format="%Y-%U-%w",
+    )
+
+    # Rename the region field to match the `state` field present in other data frames.
+    df = df.rename(columns={"REGION": STATE_FIELD})
```
Comment (lucasmbrown) [inline covid/transform.py:546]: nice!

**Review by lucasmbrown — APPROVED** [review]

**Pull request merged**